Bottom: 8919d8b519faa99e7c6eeaf55e257ba9d229c60c
Top:    66e2b7d5e12aaa88b7e70d004fe368f8394465bc
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2014-06-14 00:27:20 +0100

Refresh of add-support-for-arc-segments

---

diff --git a/src/polygon.c b/src/polygon.c
index 13aac8e..c544ac6 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -1373,7 +1373,8 @@ RemoveExcessPolygonPoints (LayerType *Layer, PolygonType *Polygon)
       line.Point1 = Polygon->Points[prev];
       line.Point2 = Polygon->Points[next];
       line.Thickness = 0;
-      if (IsPointOnLine (p->X, p->Y, 0.0, &line))
+      if (Polygon->Points[prev].included_angle == Polygon->Points[n].included_angle &&
+          IsPointOnLine (p->X, p->Y, 0.0, &line))
         {
           RemoveObject (POLYGONPOINT_TYPE, Layer, Polygon, p);
           changed = true;
